File: BEmu_cpp/src/BloombergTypes/Constant.cpp
//------------------------------------------------------------------------------------------------
// <copyright project="BEmu_cpp" file="src/BloombergTypes/Constant.cpp" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------------------------

#include "BloombergTypes/Constant.h"
#include "Exceptions/Exception.h"

namespace BEmu
{
	Constant::Constant(const Constant& original) { throw BEmu::Exception("not implemented"); }

	void setUserData(void *userData);

	Name Constant::name() const { throw BEmu::Exception("not implemented"); }
	const char * Constant::description() const { throw BEmu::Exception("not implemented"); }
	int Constant::status() const { throw BEmu::Exception("not implemented"); }
	int Constant::datatype() const { throw BEmu::Exception("not implemented"); }
	int Constant::getValueAs(char *buffer) const { throw BEmu::Exception("not implemented"); }
	int Constant::getValueAs(int *buffer) const { throw BEmu::Exception("not implemented"); }
	int Constant::getValueAs(long *buffer) const { throw BEmu::Exception("not implemented"); }
	int Constant::getValueAs(float *buffer) const { throw BEmu::Exception("not implemented"); }
	int Constant::getValueAs(double *buffer) const { throw BEmu::Exception("not implemented"); }
	int Constant::getValueAs(Datetime *buffer) const { throw BEmu::Exception("not implemented"); }
	int Constant::getValueAs(std::string *buffer) const { throw BEmu::Exception("not implemented"); }
	char Constant::getValueAsChar() const { throw BEmu::Exception("not implemented"); }
	int Constant::getValueAsInt32() const { throw BEmu::Exception("not implemented"); }
	long Constant::getValueAsInt64() const { throw BEmu::Exception("not implemented"); }
	float Constant::getValueAsFloat32() const { throw BEmu::Exception("not implemented"); }
	double Constant::getValueAsFloat64() const { throw BEmu::Exception("not implemented"); }
	Datetime Constant::getValueAsDatetime() const { throw BEmu::Exception("not implemented"); }
	std::string Constant::getValueAsString() const { throw BEmu::Exception("not implemented"); }
	void * Constant::userData() const { throw BEmu::Exception("not implemented"); }
}

File: BEmu/BloombergTypes/Constant.cs
//------------------------------------------------------------------------------
// <copyright project="BEmu_csh" file="BloombergTypes/Constant.cs" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

namespace Bloomberglp.Blpapi
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;
    using System.IO;

    public class Constant
    {
        private Constant() { }

        public Schema.Datatype Datatype { get { throw new NotImplementedException(); } }
        public string Description { get { throw new NotImplementedException(); } }
        public Name Name { get { throw new NotImplementedException(); } }
        public Schema.Status Status { get { throw new NotImplementedException(); } }
        
        public object UserData
        {
            get { throw new NotImplementedException(); }
            set { throw new NotImplementedException(); }
        }

        public char GetValueAsChar() { throw new NotImplementedException(); }
        [Obsolete("Deprecated since 3.2.1. Use GetValueAsDatetime() instead")]
        public Datetime GetValueAsDate() { throw new NotImplementedException(); }
        public Datetime GetValueAsDatetime() { throw new NotImplementedException(); }
        public float GetValueAsFloat32() { throw new NotImplementedException(); }
        public double GetValueAsFloat64() { throw new NotImplementedException(); }
        public int GetValueAsInt32() { throw new NotImplementedException(); }
        public long GetValueAsInt64() { throw new NotImplementedException(); }
        public string GetValueAsString() { throw new NotImplementedException(); }
        [Obsolete("Deprecated since 3.2.1. Use GetValueAsDatetime() instead")]
        public Datetime GetValueAsTime() { throw new NotImplementedException(); }
        public void Print(Stream output) { throw new NotImplementedException(); }
        public void Print(TextWriter writer) { throw new NotImplementedException(); }
    }
}